hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/HoodieBucketIndex.java

```diff
@@ -45,12 +48,14 @@ public abstract class HoodieBucketIndex extends HoodieIndex<Object, Object> {
 
   private static final Logger LOG = LogManager.getLogger(HoodieBucketIndex.class);
 
-  protected final int numBuckets;
-  protected final List<String> indexKeyFields;
+  protected int numBuckets;
+  protected List<String> indexKeyFields;
 
   public HoodieBucketIndex(HoodieWriteConfig config) {
     super(config);
-
+    if (config.getBucketIndexEngineType() == BucketIndexEngineType.RANGE_BUCKET) {
+      return;
+    }
```

Comment:
We should not write sub-class specific logic in base classes. So let's not return early here.